Robot_Adapter/CRUD/Read/Elements/Meshes.cs

```diff
@@ -145,11 +148,56 @@ private List<FEMesh> ReadMeshes(List<string> ids = null)
                     bhomMeshes[panelId].Nodes.Add(bhomNodes[nodeId]);
                 }
             }
+
+            RobotObjObjectServer robotObjectServer = m_RobotApplication.Project.Structure.Objects;
+
+            foreach (KeyValuePair<int, FEMesh> kvp in bhomMeshes)
+            {
+                int id = kvp.Key;
+                FEMesh mesh = kvp.Value;
+
+                //Get local orientations for each face
+                List<Vector> normals = mesh.Normals();
+
+                //Check if all orientations are the same
+                bool sameOrientation = true;
+
+                for (int i = 0; i < normals.Count - 1; i++)
+                {
+                    sameOrientation &= normals[i].Angle(normals[i + 1]) < Tolerance.Angle;
+                }
+
+                if (sameOrientation)
+                {
+                    Vector normal = normals.First();
+                    RobotObjObject robotPanel = robotObjectServer.Get(id) as RobotObjObject;
+
+                    double x, y, z; robotPanel.Main.Attribs.GetDirX(out x, out y, out z);
+
+                    bool flip = robotPanel.Main.Attribs.DirZ == 1;
+                    flip = Convert.FromRobotCheckFlipNormal(normal, flip);
+
+                    if (flip)
+                    {
+                        normal = normal.Reverse();
```

Comment:
*-1 could allow avoiding cloning if I understand the code correctly

Reply:
done